### DISCUSSION TITLE
How to modify /usr/local/apisix/conf/nginx.conf

### DISCUSSION THREAD

**Discussion by xiongyunhua-star** [apache/apisix#5064] [Q&A]
### Issue description

I want to insert a header （k : v）in nginx.conf of APISIX

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by xiongyunhua-star** [comment]
But the file is read-only

**Comment by tzssangglass** [comment]
take a look at:https://github.com/apache/apisix/blob/master/docs/zh/latest/customize-nginx-configuration.md

**Comment by nic-chen** [comment]
@xiongyunhua-star 

You could also see if this plugin can meet your needs:
https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/proxy-rewrite.md

**Comment by tokers** [comment]
1. Users should avoid to modify the `nginx.conf` directly because it's generated by APISIX, your modifications will be overwritten, instead, changing `apisix/cli/ngx_tpl.lua` if necessary;
2. The configuration was mount from a ConfigMap, and is read-only, so you cannot modify this file, if desired, modify the mount mode from RO to RW.
